### PR TITLE
add inspector command for system inspection

### PR DESCRIPTION
**Pull request by frrist**
### Description
This PR makes progress towards completing #2652  by adding the following commands:
```
USAGE
  go-filecoin inspect - Show info about the filecoin node

  go-filecoin inspect

SUBCOMMANDS
  go-filecoin inspect config  - Print in memory config information.
  go-filecoin inspect disk    - Print filesystem usage information.
  go-filecoin inspect memory  - Print memory usage information.
  go-filecoin inspect runtime - Print runtime diagnostic information.

Use 'go-filecoin inspect --help' for more information about this command.
```

**Disk**:  Prints out information about the filesystem
```
$ go-filecoin inspect disk
{
  "FSType": "61267",
  "Free": 860286201856,
  "Total": 809074733056
}
```

**Memory**: Prints out information about system memory.
```
$ go-filecoin inspect memory
{
  "Swap": 0,
  "Virtual": 10323220000
}
```

**Runtime**:  Prints out information about your golang runtime.
```
$ go-filecoin inspect runtime
{
  "Arch": "amd64",
  "Compiler": "gc",
  "GoMaxProcs": 8,
  "NumCGoCalls": 11,
  "NumGoRoutines": 90,
  "NumProc": 8,
  "OS": "linux",
  "Version": "go1.12.1"
}
```

**Environment**:  Prints out information about your filecoin nodes environment.
```
$ go-filecoin inspect environment
{
  "FIL_API": "",
  "FIL_PATH": "",
  "GOPATH": "/home/frrist"
}
```

**Config**: Prints out information about your filecoin nodes in memory config.
```
go-filecoin inspect config
{
  "api": {
    "accessControlAllowCredentials": false,
    "accessControlAllowMethods": [
      "GET",
      "POST",
      "PUT"
    ],
....omitted....
  "wallet": {
    "defaultAddress": "t1azpabiyccur7ojhicilqti3ddjvtkbopa42s7py"
  }
} 
```

### CODE REVIEW

File: commands/diagnostic.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	var straddrs []string
+	for _, a := range addrs {
+		straddrs = append(straddrs, a.String())
+	}
+
+	out["interface_addresses"] = straddrs
+	return nil
+}
+
+func filPath() string {
```
Comment (ZenGround0) [inline commands/diagnostic.go]: Blocking: This won't work if a user specified home directory with the `--repodir` flag, you should parse that arg, import `repo` and use `repo.GetRepoDir(repoDirArgString)` function as in other commands.  Also this way the env var stuff is taken care of for you.

**Comment by mishmosh** [comment]
Are `diag` and `sys` fairly universal command names? If not, could they be renamed to be more human-readable?

**Comment by frrist** [comment]
@mishmosh I was originally borrowing from the naming [convention used by go-ipfs](https://github.com/ipfs/go-ipfs/blob/master/core/commands/diag.go#L8), however I have altered this in attempt to be a bit more descriptive. The PR description now reflects this.

@ZenGround0 I have re-written this inlight of your review to use plumbing, I think `stats` could be moved under the `inspect` command -- could you give this a second?

File: plumbing/inspector/service.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+		repo: r,
+	}
+}
+
+// Service contains information used to inspect the go-filecoin node.
+type Service struct {
+	repo repo.Repo
+}
+
+// RuntimeInfor contains information about the golang runtime.
```
Comment (travisperson) [inline plumbing/inspector/service.go]: > RuntimeInfo**r**

File: plumbing/inspector/service.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	Version       string
+	Compiler      string
+	NumProc       int
+	GoMaxProcs    int
+	NumGoRoutines int
+	NumCGoCalls   int64
+}
+
+// RuntimeInfor contains information about the environment filecoin is running in.
+type EnvironmentInfo struct {
```
Comment (travisperson) [inline plumbing/inspector/service.go]: > **EnvironmentInfo** contains

File: plumbing/inspector/service.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+		Version:       runtime.Version(),
+		Compiler:      runtime.Compiler,
+		NumProc:       runtime.NumCPU(),
+		GoMaxProcs:    runtime.GOMAXPROCS(0),
+		NumGoRoutines: runtime.NumGoroutine(),
+		NumCGoCalls:   runtime.NumCgoCall(),
+	}
+}
+
+func (g *Service) Environment() *EnvironmentInfo {
```
Comment (travisperson) [inline plumbing/inspector/service.go]: Need a doc comment

File: repo/fsrepo.go [hunk 5]
```diff
@@ -512,6 +512,10 @@ func (r *FSRepo) APIAddr() (string, error) {
 	return APIAddrFromFile(filepath.Join(filepath.Clean(r.path), APIFile))
 }
 
+func (r *FSRepo) Path() string {
```
Comment (travisperson) [inline repo/fsrepo.go]: Need a doc comment

**Comment by anorth** [comment]
Do we want an `inspect all` so there's a one-liner for our bug reports?

**Comment by frrist** [comment]
>  The alternative here is to put the inspector service on the command Env

This is a much better idea imo, I have modified it to do such.

> Do we want an inspect all so there's a one-liner for our bug reports?

Done:
```json
go-filecoin inspect all | jq
{
  "Config": {
    "api": {
      "accessControlAllowCredentials": false,
      "accessControlAllowMethods": [
        "GET",
        "POST",
        "PUT"
      ],
      "accessControlAllowOrigin": [
        "http://localhost:8080",
        "https://localhost:8080",
        "http://127.0.0.1:8080",
        "https://127.0.0.1:8080"
      ],
      "address": "/ip4/127.0.0.1/tcp/3453"
    },
    "bootstrap": {
      "addresses": [],
      "minPeerThreshold": 0,
      "period": "1m"
    },
    "datastore": {
      "path": "badger",
      "type": "badgerds"
    },
    "heartbeat": {
      "beatPeriod": "3s",
      "beatTarget": "",
      "nickname": "",
      "reconnectPeriod": "10s"
    },
    "metrics": {
      "prometheusEnabled": false,
      "prometheusEndpoint": "/ip4/0.0.0.0/tcp/9400",
      "reportInterval": "5s"
    },
    "mining": {
      "autoSealIntervalSeconds": 120,
      "minerAddress": "empty",
      "storagePrice": "0"
    },
    "mpool": {
      "maxNonceGap": "100",
      "maxPoolSize": 10000
    },
    "net": "",
    "sectorbase": {
      "rootdir": ""
    },
    "swarm": {
      "address": "/ip4/0.0.0.0/tcp/6000"
    },
    "wallet": {
      "defaultAddress": "t1ciamdwkjhbt7fjqm4chc3logy76cjbjey75qfia"
    }
  },
  "Disk": {
    "FSType": "61267",
    "Free": 856211677184,
    "Total": 805000208384
  },
  "Environment": {
    "FIL_API": "",
    "FIL_PATH": "",
    "GOPATH": "/home/frrist"
  },
  "Memory": {
    "Swap": 0,
    "Virtual": 10331684000
  },
  "Runtime": {
    "Arch": "amd64",
    "Compiler": "gc",
    "GoMaxProcs": 8,
    "NumCGoCalls": 9,
    "NumGoRoutines": 87,
    "NumProc": 8,
    "OS": "linux",
    "Version": "go1.12.1"
  }
}
```

**Comment by frrist** [comment]
> Please follow up with an update to the GitHub bug issue template so we can see this in action.

Sounds good, will do this once it merges

**Pull request merged**